src/write_outputs/write_outputs.jl

```diff
@@ -581,5 +584,190 @@ function write_full_time_series_reconstruction(
+
+    The keyword arguments `filetype` and `compression` are optional and are both set to `auto_detect` by default.
+
+    # Arguments
+    - `path::AbstractString`: The path including the file name. This can include the file extension (e.g. .csv) but does not have to.
+    - `file::DataFrame`: The DataFrame being saved to the input path. All columns in the DataFrame must have a type (cannot be type "Any") in order for DuckDB to work.
+    - `filetype::String`: The file type, as specified in `ResultsFileType` in `genx_settings.yml`. Accepted inputs are `.csv`,`.csv.gz` `.parquet`, `.json`, `.json.gz`, and `auto_detect` (default).
+    - `compression::String`: The compression type, as specified in `ResultsCompressionType` in `genx_settings.yml`. Accepted inputs are `gzip`, `snappy`, `zstd`, `none`, and `auto_detect` (default). 
+"""
+function write_output_file(path::AbstractString, file::DataFrame; filetype::String = "auto_detect", compression::String = "auto_detect")
+    # 1) Create a DuckDB Interface
+    con = DBInterface.connect(DuckDB.DB)
+    println(typeof(file))
+    println(file)
+
+    # 2) Register the DataFrame with DuckDB
+    DuckDB.register_data_frame(con, file, "temp_df")
+
+    # 3) Check if an extension is already in the file name, if not, add it based on filetype
+    if occursin(".", path)
+        if occursin(".", splitext(path)[1]) # If two extensions are present (eg .csv.gz, or .json.gz, only the first will be added to the filetype as .gz will be autodetected by DuckDB later)
+            if filetype == "auto_detect" # If auto-detect is on for the extension type, change the filetype to the extension detected using splitext
+                filetype = splitext(splitext(path)[1])[2]
+            elseif filetype != splitext(splitext(path)[1])[2] # If the extension in the file name is different than the filetype key, override the filetype key and throw a warning.
+                newfiletype = splitext(path)[2]
+                @warn("Filetype '$filetype' is incompatible with extension specified in results_settings.yml. Saving as '$newfiletype' instead.")
+                filetype = splitext(splitext(path)[1])[2]
+            end
+        else
+            if filetype == "auto_detect"  # If auto-detect is on for the extension type, change the filetype to the extension detected using splitext
+                filetype = splitext(path)[2]
+            elseif filetype != splitext(path)[2] # If the extension in the file name is different than the filetype key, override the filetype key and throw a warning.
+                newfiletype = splitext(path)[2]
+                @warn("Filetype '$filetype' is incompatible with extension specified in results_settings.yml. Saving as '$newfiletype' instead.")
+                filetype = splitext(path)[2]
+            end
+            if splitext(path)[2] == ".csv" && (compression == ".gz" || compression == "gz" || compression == "gzip") 
```

Comment:
This comparison
`(compression == ".gz" || compression == "gz" || compression == "gzip")`
gets made a bunch of times. Consider 

a) normalize `compression` internally so that if the function is called with ".gz" or "gz" or "gzip" it's always "gzip", or something.
b) save the result to a boolean `is_gzip` or something.